### DISCUSSION TITLE
type:SELF 수정후 oAuth2.0 오류 문의

### DISCUSSION THREAD

**Discussion by hoksi** [commerce-api-naver/commerce-solutions#76] [묻고 답하기]
- gid : cr3-150454-65772h^1732687548282^48523971
- 오류메시지
`{"code":"DataAccessDenied","message":"데이터에 대한 접근 권한이 없습니다.","timestamp":"2024-12-02T02:43:15.636150Z","invalidInputs":[{"name":"account_id","type":"NotAuthority.subscription","message":"subscription not found matching applicationId and accountId. CODE : SUBSCRIPTION_NOT_FOUND"}]}` 

사용 시작 승인을 위해 oAuth2.0 인증을 요청하는데 오류가 발생합니다.
SELLER의 경우 "스마트스토어 계정 UID" 계정이 필요하다 하셨는데 어떻게 확인이 가능할까요?
아직 출시되지 않은 솔루션 개발진행중입니다.

**Comment by commerce-api-naver** [comment]
@hoksi 

안녕하세요 커머스API 권현철입니다.

close 이후 언급이 없으셔서 답변 드립니다.

'인증 토큰 발급 요청API' 호출 간 솔루션을 구독하지 않은 사용자의 정보를 입력하여 해당 오류가 발생하였습니다.
account_id는 솔루션의 구독을 신청한 판매자 또는 구독중인 판매자의 스마트스토어 계정 UID를 사용해야 정상 호출이 가능합니다.

스마트스토어 계정 UID는 솔루션 유형에 따라 다르게 식별합니다.
- '바로 사용' 유형: JWT로 구독자 식별
    -  JWT는 커머스솔루션마켓 또는 스마트스토어센터 솔루션목록에서 판매자가 [신청하기] 또는 [사용하기] 버튼 클릭 시 솔루션 페이지로 이동하며 전달됩니다.
- '심사 후 승인': 판매자 인증 JWE는 토큰 형태로, JWE 해석 API를 호출하여 분석 가능한 형식으로 변환해야 판매자의 솔루션 구독 정보를 확인할 수 있으며 'JWE 해석 API'는 대상 솔루션 애플리케이션 ID로만 정상 해석 결과 확인이 가능합니다.

각 유형의 연동 가이드는 하단의 커머스API센터 기본 연동 요소 가이드를 참고 부탁드립니다.

- https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EA%B0%9C%EB%B0%9C%ED%95%98%EA%B8%B0/%EA%B8%B0%EB%B3%B8-%EC%97%B0%EB%8F%99-%EC%9A%94%EC%86%8C-%EA%B0%80%EC%9D%B4%EB%93%9C (커머스API센터 기본 연동 요소 가이드)

감사합니다.
